website/docs/internals/json-format.mdx

```diff
@@ -307,6 +308,8 @@ The following example illustrates the structure of a `<values-representation>`:
 
 The translation of attribute and output values is the same intuitive mapping from HCL types to JSON types used by Terraform's [`jsonencode`](/language/functions/jsonencode) function. This mapping does lose some information: lists, sets, and tuples all lower to JSON arrays while maps and objects both lower to JSON objects. Unknown values and null values are both treated as absent or null.
 
+Output values include a `"type"` field, which is a [serialization of the value's type](https://pkg.go.dev/github.com/zclconf/go-cty/cty#Type.MarshalJSON). For primitive types this is a string value, such as `"number"` or `"bool"`. Complex types are represented as a nested JSON array, such as `["map","string"]` or `["object",{"a":"number"]]`. This can be used to reconstruct the output value with the correct type.
+
```

Comment:
Probably we should aim to inline the information about this into this document rather than referring to the external dependency that happens to implement it today, but this does seem like a reasonable compromise to allow us to get moving on this relatively quickly and improve the docs separately later.